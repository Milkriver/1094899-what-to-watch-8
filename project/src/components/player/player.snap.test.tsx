import { render } from '@testing-library/react';
import { configureMockStore } from '@jedmao/redux-mock-store';
import Player from './player';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';


describe('Component: Player', () => {
  const mockStore = configureMockStore();
  const store = mockStore({
    MOVIES: {
      movie: {
        id: 1, name: '', posterImage: '', previewImage: '', backgroundImage: '',
        backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 1, scoresCount: 1, director: '', starring: ['', ''],
        runTime: 1, genre: '', released: 1, isFavorite: true,
      },
      activeMovie: {
        id: 1, name: '', posterImage: '', previewImage: '', backgroundImage: '',
        backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 1, scoresCount: 1, director: '', starring: ['', ''],
        runTime: 1, genre: '', released: 1, isFavorite: true,
      },
    },
    USER: {},
    REVIEWS: {},
  });

  HTMLMediaElement.prototype.play = jest.fn();
  HTMLMediaElement.prototype.pause = jest.fn();
  const history = createMemoryHistory();
  test('should render correctly', () => {
    const page = (
      <Provider store={store}>
        <Router history={history}>
          <Player />
        </Router>
      </Provider>
    );
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
