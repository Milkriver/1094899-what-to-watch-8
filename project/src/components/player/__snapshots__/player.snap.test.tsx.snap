// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Player should render correctly 1`] = `
<div>
  <div
    class="player"
  >
    <div
      style="display: flex; justify-content: center; height: 100vh; align-items: center;"
    >
      Загрузка...
    </div>
    <video
      class="player__video"
      src=""
    />
    <a
      href="/films/1"
    >
      <button
        class="player__exit"
        type="button"
      >
        Exit
      </button>
    </a>
    <div
      class="player__controls"
    >
      <div
        class="player__controls-row"
      >
        <div
          class="player__time"
        >
          <progress
            class="player__progress"
            max="0"
            value="0"
          />
          <div
            class="player__toggler"
          >
            Toggler
          </div>
        </div>
        <div
          class="player__time-value"
        >
          - 00:00
        </div>
      </div>
      <div
        class="player__controls-row"
      >
        <button
          class="player__play"
          disabled=""
          type="button"
        >
          <svg
            height="21"
            viewBox="0 0 14 21"
            width="14"
          >
            <use
              xlink:href="#pause"
            />
          </svg>
          <span>
            Play
          </span>
        </button>
        <div
          class="player__name"
        />
        <button
          class="player__full-screen"
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlink:href="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
